### PR TITLE
Set default values for vxlanVNI and BPFHostConntrackBypass for DockerEE

### PR DESCRIPTION
**Pull request by sridhartigera**
## Description

1. DockerEE uses vxlanPort 4789 and this results in a conflict. Hence changing the default vxlanVNI for DockerEE to 10000.
2. When in BPF mode in DockerEE, ```BPFHostConntrackBypass``` needs to be disabled for cluster connectivity to work. 

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by tmjd** [comment]
Is there a specific reason 4798 was chosen? It looks very similar to the value we're trying to avoid and might look like a typo. WDYT about using 4790, I think that is a little better.

**Comment by sridhartigera** [comment]
> Is there a specific reason 4798 was chosen? It looks very similar to the value we're trying to avoid and might look like a typo. WDYT about using 4790, I think that is a little better.



> Is there a specific reason 4798 was chosen? It looks very similar to the value we're trying to avoid and might look like a typo. WDYT about using 4790, I think that is a little better.

Might end up changing the VNI. Will update the PR accordingly. MKE uses ```vni:4096, port:4789``` for their swarm overlay. We thought of using a different port but shaun thinks its better to use a different VNI. Will update the PR accordingly.

File: pkg/controller/installation/core_controller.go [hunk 1]
```diff
@@ -1686,6 +1686,20 @@ func (r *ReconcileInstallation) setDefaultsOnFelixConfiguration(ctx context.Cont
...
+	// This results in a conflict with calico's VXLAN and the vxlan.calico interface
+	// gets deleted. To fix this we change the vxlanVNI to 10000 as recommended by
+	// MKE docs (https://docs.mirantis.com/mke/3.7/cli-ref/mke-cli-install.html).
+	if install.Spec.KubernetesProvider == operator.ProviderDockerEE {
+		// Set bpfHostConntrackBypass to false for eBPF dataplane to work with MKE
+		if install.Spec.BPFEnabled() {
+			disableBPFHostConntrackBypass(fc)
+		}
+		vxlanVNI = 10000
+		fc.Spec.VXLANVNI = &vxlanVNI
```
Comment (tmjd) [inline pkg/controller/installation/core_controller.go:1699]: The code here should only be setting values if they are not already set.
That will also hopefully ensure that this change won't impact functionality in existing clusters.

Does `fc.Spec.VXLANVNI` get defaulted by CalicoNode/Felix? If not that could be an issue with impacting existing clusters, unless it is safe to change VXLANVNI on the fly.

**Review by tmjd — APPROVED** [review]
LGTM

**Review by fasaxc — APPROVED** [review]

**Pull request merged**